src/ExpressiveInstaller/Resources/src/ExpressiveSymfonyContainer.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace App;
+
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Dumper\PhpDumper;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Configuration for the Symfony Dependency Injection container.
+ *
+ * This class provides functionality for the following service types:
+ *
+ * - Aliases
+ * - Delegators
+ * - Factories
+ * - Invokable classes
+ * - Services (known instances)
+ */
+class ExpressiveSymfonyContainer
+{
+    const ENABLE_CACHE = 'cache_symfony_container';
+
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var string
+     */
+    private $cachedContainerFile;
+
+    /**
+     * @param array $config
+     * @param string $cachedContainerFile file name to use to store the cached container
+     */
+    public function __construct(
+        array $config,
+        $cachedContainerFile
+    ) {
+        $this->config = $config;
+        $this->cachedContainerFile = $cachedContainerFile;
+    }
+
+    /**
+     * @return Container
+     */
+    public function create()
+    {
+        $cacheEnabled = file_exists($this->cachedContainerFile) &&
+            isset($this->config[static::ENABLE_CACHE]) &&
+            $this->config[static::ENABLE_CACHE];
+
+        if (! $cacheEnabled) {
+            $this->cacheContainer($this->build());
+        }
+
+        $container = $this->loadFromCache($this->cachedContainerFile);
+
+        // Inject config
+        $container->set('config', $this->config);
```

Comment:
Should the config array really be set as a service (btw, arrays are not really meant to be added as services in Symfony, but as parameters).
And does `config['dependencies']` really need to be set here ? It is processed to build the container.

Btw, you could even set the config at build time instead of injecting it at runtime (unless you invalidate the cache only when `config['dependencies']` changes rather than the whole config. But I don't know what else you could have in this config)